packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2927,9 +2930,34 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           this.resetStartLocationFromNode(expr, typeParameters);
         }
         expr.typeParameters = typeParameters;
+
+        // report error if single type parameter used without trailing comma.
+        if (
+          this.hasPlugin("jsx") &&
+          expr.typeParameters.params.length === 1 &&
+          !expr.typeParameters.extra?.trailingComma
+        ) {
+          const parameter = expr.typeParameters.params[0];
+          if (parameter.constraint) {
+            // If parameter has any constraints, it must contain multiple tokens.
+            // <T extends U> is a valid declaration.
+            // <T extends {name: string}> is also a valid declaration.
+          } else invalidSingleType = parameter;
+        }
+
         return expr;
       }, state);
 
+      if (invalidSingleType) {
+        this.raise(
+          invalidSingleType.end + 1,
```

Comment:
In https://github.com/babel/babel/commit/478a9709ab20dc942fbb2c20a82ae8fa41e55ed6 we changed the signature of `.raise`; could you rebase this PR? :pray:  (otherwise I can do it)

As the new location, you can probably use `{ at: this.state.startLoc }` (which is the beginning of the token after the type parameter).

Reply:
Hi @nicolo-ribaudo , of course I can do it. I Will do it today night. But if you have any urgency, you can do it for me as well.